pkg/sql/logictest/testdata/logic_test/partial_index

```diff
@@ -455,3 +459,136 @@ SELECT * FROM d@f_b_s_foo_idx WHERE b AND s = 'foo'
 100  110  100  foo  true
 300  330  300  foo  true
 400  400  400  foo  true
+
+#### Backfill a partial index.
+
+statement ok
+CREATE TABLE e (a INT, b INT)
+
+statement ok
+INSERT INTO e VALUES (1, 10);
+INSERT INTO e VALUES (2, 20);
+INSERT INTO e VALUES (3, 30);
+INSERT INTO e VALUES (4, 40);
+INSERT INTO e VALUES (5, 50);
+INSERT INTO e VALUES (6, 60);
+
+statement ok
+CREATE INDEX a_b_gt_30_idx ON e (a) WHERE b > 30
+
+query II
+SELECT * FROM e@a_b_gt_30_idx WHERE b > 30
```

Comment:
Is there some way that we can scan the index manually to ensure that there aren't extra KV's? Maybe something like the tests in `schema_changer_test.go`

Reply:
Forcing the index should be the same thing, AFAIK. This test will fail if a row is added to the partial index when it shouldn't be.

Reply:
I've added a comment to make it more clear that this will be a full partial index scan.